components/moderation-panel/ModEditSubmissionForm.vue

```diff
@@ -456,6 +458,74 @@ function initializeSubmissionFormValues(submissionData: Submission | undefined)
+        const snapshotBase: Submission = {
+            ...submissionData,
+            healthcareProfessionalName: submissionData.healthcareProfessionalName ?? '',
+            googleMapsUrl: submissionData.googleMapsUrl ?? '',
+            spokenLanguages: submissionData.spokenLanguages ? [...submissionData.spokenLanguages] : []
+        }
+
+        // Reconstruct the FacilitySubmission object from the current form field values
+        // (facilitySectionFields) instead of using submissionData.facility directly.
+        // This ensures that we capture the latest edited state for comparison or submission.
+        const reconstructedFacility: FacilitySubmission = {
+            id: submissionData?.facility?.id || '',
+            nameEn: facilitiesStore.facilitySectionFields.nameEn || '',
+            nameJa: facilitiesStore.facilitySectionFields.nameJa || '',
+            mapLatitude: parseFloat(facilitiesStore.facilitySectionFields.mapLatitude) || 0,
+            mapLongitude: parseFloat(facilitiesStore.facilitySectionFields.mapLongitude) || 0,
+            contact: {
+                phone: facilitiesStore.facilitySectionFields.phone || '',
+                email: facilitiesStore.facilitySectionFields.email || '',
+                website: facilitiesStore.facilitySectionFields.website || '',
+                googleMapsUrl: facilitiesStore.facilitySectionFields.googlemapsURL || '',
+                address: {
+                    postalCode: facilitiesStore.facilitySectionFields.postalCode || '',
+                    prefectureEn: facilitiesStore.facilitySectionFields.prefectureEn || '',
+                    cityEn: facilitiesStore.facilitySectionFields.cityEn || '',
+                    addressLine1En: facilitiesStore.facilitySectionFields.addressLine1En || '',
+                    addressLine2En: facilitiesStore.facilitySectionFields.addressLine2En || '',
+                    prefectureJa: facilitiesStore.facilitySectionFields.prefectureJa || '',
+                    cityJa: facilitiesStore.facilitySectionFields.cityJa || '',
+                    addressLine1Ja: facilitiesStore.facilitySectionFields.addressLine1Ja || '',
+                    addressLine2Ja: facilitiesStore.facilitySectionFields.addressLine2Ja || '',
+                    __typename: 'PhysicalAddress' as const
+                },
+                __typename: 'Contact' as const
+            },
+            healthcareProfessionalIds: [...facilitiesStore.facilitySectionFields.healthcareProfessionalIds],
+            __typename: 'FacilitySubmission' as const
+        }
+
+        // Reconstruct the HealthcareProfessionalSubmission object from the current form field values
+        // (healthcareProfessionalSections) instead of using submissionData.healthcareProfessional directly.
+        // This ensures that we capture the latest edited state for comparison or submission.
+        const reconstructedHealthcareProfessionals: HealthcareProfessionalSubmission[] = [{
```

Comment:
```suggestion
        const reconstructedHealthcareProfessionalSubmission: HealthcareProfessionalSubmission[] = [{
```
I removed the s as it mat ch ed the type and follows the variable understanding first concept